### ISSUE TITLE
Dealing with Fake IP

### ISSUE THREAD

**Issue by glfruit** [wallabag/wallabag#6137]
**Before submitting the issue, please read:**
If wallabag can't parse / extract content for a given link, please first read the documentation about it:
http://doc.wallabag.org/en/user/errors_during_fetching.html#how-can-i-help-to-fix-that

We have a lot of requests about fetching config issue. It'll help us A LOT if you give a try to fix it on your own following the doc.
If you failed to fix it yourself, tick the following boxes:
- [ ] I've tried myself without success
- [ ] I've replaced `HOST` in the issue title with the host of the URL that can't be fetched (ie: `nytimes.com`, `20minutes.fr`, `bbc.com`, etc.)

**Content related:**
 - URL: [full url of the content]
 - wallabag version: [e.g. 2.4.0]

**Describe what's wrong:**
I run wallabag behind a Clash proxy, the problem is Clash always returns fake ip for any domain which wallbag uses to fetch content. Obviously this can't work because fake ip is just a private network address. Is there any way to make wallbag work properly with Clash?

**Comment by 9k001** [comment]
I have the same problem as you. Have you made any progress on this problem?

**Comment by 9k001** [comment]
After my constant search, I found the cause of the problem.
At the beginning of the period, a discussion of security took place here. [Server Side Request Forgery (SSRF) ](https://github.com/j0k3r/graby/issues/7)
The author of wallabag wrote a security plug-in . [httplug-ssrf-plugin](https://github.com/j0k3r/httplug-ssrf-plugin.git)
Among them, some blocked IP are written here. [Options.php](https://github.com/j0k3r/httplug-ssrf-plugin/blob/master/src/Options.php) 
So, my solution is to comment out the settings about FakeIP.
```
# cp /var/www/wallabag/vendor/j0k3r/httplug-ssrf-plugin/src/Options.php /var/www/wallabag/vendor/j0k3r/httplug-ssrf-plugin/src/Options.php.bak
# vi /var/www/wallabag/vendor/j0k3r/httplug-ssrf-plugin/src/Options.php

        'blacklist' => [
            'ip' => [
                '0.0.0.0/8',
                '10.0.0.0/8',
                '100.64.0.0/10',
                '127.0.0.0/8',
                '169.254.0.0/16',
                '172.16.0.0/12',
                '192.0.0.0/29',
                '192.0.2.0/24',
                '192.88.99.0/24',
                '192.168.0.0/16',
                /**'198.18.0.0/15',*/
                '198.51.100.0/24',
                '203.0.113.0/24',
                '224.0.0.0/4',
                '240.0.0.0/4',
            ],

```

But I still want @j0k3r  to pay attention to this problem. Modify the relevant code. After all, mounting files is only a temporary solution.

**Comment by j0k3r** [comment]
Yeah maybe we should allow end user to configure the blacklist ip array.

**Comment by 9k001** [comment]
> Yeah maybe we should allow end user to configure the blacklist ip array.

https://github.com/wallabag/wallabag/issues/6137#issuecomment-1969235594

**Comment by 9k001** [comment]
> Yeah maybe we should allow end user to configure the blacklist ip array.

I think having a switch is a better choice. If it is a configuration item, there may be problems with input parameters.

**Comment by 9k001** [comment]
> What do you mean by _having a switch_?

Well, I use translation software. The switch has a button, which is opened to add the security setting, and off to cancel the security setting.
